src/Grace/DependencyInjection/Impl/DisposalScope.cs

```diff
@@ -1,8 +1,231 @@
 ﻿using System;
 using System.Threading;
+#if NETSTANDARD2_1
+using System.Threading.Tasks;
+#endif
 
 namespace Grace.DependencyInjection.Impl
 {
+#if NETSTANDARD2_1
```

Comment:
It feels like you're replicating a lot of code by putting the if here vs. having multiple if statements nested in the code (not ideal but seems better than duplicating some of the implementation).

Reply:
Done!